sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -0,0 +1,560 @@
+  let studentAppealRepo: Repository<StudentAppeal>;
+  let formService: FormService;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    appModule = module;
+    applicationRepo = dataSource.getRepository(Application);
+    studentAppealRequestRepo = dataSource.getRepository(StudentAppealRequest);
+    studentAppealRepo = dataSource.getRepository(StudentAppeal);
+    student = await saveFakeStudent(appDataSource);
+    studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Setup form service to mock form methods
+    formService = await getProviderInstanceForModule(
+      appModule,
+      AppStudentsModule,
+      FormService,
+    );
+    // Mock user service to return the saved student.
+    mockUserLoginInfo(appModule, student);
+  });
+
+  it("Should throw not found error when the application does not exist.", async () => {
+    // Arrange
+    const endpoint = `/students/appeal/application/123456`;
+    const payload = {
+      studentAppealRequests: [{ formData: {}, formName: "test" }],
+    };
+
+    // Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .send(payload)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        message:
+          "Given application either does not exist or is not complete to request change.",
+        errorType: INVALID_APPLICATION_NUMBER,
+      });
+  });
+
+  it("Should throw unprocessable entity exception error when the application is archived.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(appDataSource, {
+      student: student,
+    });
+    const endpoint = `/students/appeal/application/${application.id}`;
+    const payload = {
+      studentAppealRequests: [{ formData: {}, formName: "test" }],
+    };
+
+    // Act
+    // Set application status to completed and archived field to true
+    application.isArchived = true;
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+
+    // Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .send(payload)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        message: "This application is no longer eligible to request changes.",
+        errorType: APPLICATION_CHANGE_NOT_ELIGIBLE,
+      });
+  });
+
+  it("Should throw unprocessable entity exception error when there is already a pending appeal.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(appDataSource, {
+      student: student,
+    });
+    const endpoint = `/students/appeal/application/${application.id}`;
+    const payload = {
+      studentAppealRequests: [{ formData: {}, formName: "test" }],
```

Comment:
I would suggest to put a very simple real time payload for formData (e.g. student answering no for has dependents) so that when we bring the dry run submission into E2E tests it will be easy.